odoo/addons/base/models/res_partner_bank.py

```diff
@@ -0,0 +1,208 @@
+import re
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+def sanitize_account_number(acc_number):
+    if acc_number:
+        return re.sub(r'\W+', '', acc_number).upper()
+    return False
+
+
+class ResPartnerBank(models.Model):
+    _name = 'res.partner.bank'
+    _rec_name = 'acc_number'
+    _description = 'Bank Account'
+    _order = 'sequence, id'
+
+    @api.model
+    def get_supported_account_types(self):
```

Comment:
Why do you need a public and a private method? Choose one (probably the private one is enough).